File: app/src/main/java/com/jonasgerdes/stoppelmap/model/transportation/Route.java
package com.jonasgerdes.stoppelmap.model.transportation;


import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Created by Jonas on 08.07.2016.
 */
public class Route extends RealmObject {

    @PrimaryKey
    private String uuid;
    private String name;
    private RealmList<Station> stations;
    private Station returnStation;

    public Route() {
    }

    public Route(String uuid, String name, RealmList<Station> stations, Station returnStation) {
        this.uuid = uuid;
        this.name = name;
        this.stations = stations;
        this.returnStation = returnStation;
    }

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public RealmList<Station> getStations() {
        return stations;
    }

    public void setStations(RealmList<Station> stations) {
        this.stations = stations;
    }

    public Station getReturnStation() {
        return returnStation;
    }

    public void setReturnStation(Station returnStation) {
        this.returnStation = returnStation;
    }

    @Override
    public String toString() {
        return "Route{" +
                "uuid=" + uuid +
                ", name='" + name + '\'' +
                ", stations=" + stations +
                ", returnStation=" + returnStation +
                '}';
    }
}

File: app/src/main/java/com/jonasgerdes/stoppelmap/model/transportation/Route.kt
package com.jonasgerdes.stoppelmap.model.transportation

import android.arch.persistence.room.*
import android.support.v7.util.DiffUtil
import com.jonasgerdes.stoppelmap.model.map.entity.Stall
import io.reactivex.Flowable

@Entity(tableName = "routes")
data class Route(
        @PrimaryKey
        val slug: String,
        val name: String
) {
    object DiffCallback : DiffUtil.ItemCallback<Route>() {
        override fun areItemsTheSame(old: Route, new: Route) = old.slug == new.slug

        override fun areContentsTheSame(old: Route, new: Route) = old == new

    }
}

@Entity(tableName = "stations")
data class Station(
        @PrimaryKey
        val slug: String,
        val name: String,
        val route: String,
        val comment: String,
        val longitude: Double,
        val latitude: Double
)

data class RouteWithStations(
        @Embedded
        val stall: Stall,
        val stations: List<Station>
)

@Dao
interface RouteDao {

    @Query("SELECT * FROM routes")
    fun getAll(): Flowable<List<Route>>
}